src/phy/loopback.rs

```diff
@@ -63,8 +63,12 @@ pub struct RxToken {
 }
 
 impl phy::RxToken for RxToken {
-    fn consume<R, F: FnOnce(&[u8]) -> Result<R>>(self, _timestamp: Instant, f: F) -> Result<R> {
-        f(&self.buffer)
+    fn consume<R, F: FnOnce(&mut [u8]) -> Result<R>>(
+        mut self,
+        _timestamp: Instant,
+        f: F
```

Comment:
This is a bit silly, but can you add a `where` clause here? I hate how you go one character beyond the line wrap and the Rust tools instantly explode everything to live on its own line.